tests/v2/validation/provisioning/k3s/etcd_node_delete_test.go

```diff
@@ -0,0 +1,221 @@
+package k3s
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+	"time"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+	nodestat "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultNS                = "default"
+	etcdLabel                = "rke.cattle.io/etcd-role"
+	clusterLabel             = "cluster.x-k8s.io/cluster-name"
+	machineSteveResourceType = "cluster.x-k8s.io.machine"
+)
+
+type EtcdNodeDeleteTestSuite struct {
+	suite.Suite
+	session               *session.Session
+	client                *rancher.Client
+	ns                    string
+	k3skubernetesVersions []string
+	cnis                  []string
+	providers             []string
+	nodesAndRoles         []machinepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.k3skubernetesVersions = clustersConfig.K3SKubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRoles
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) provisionK3SCluster(client *rancher.Client, k8sVersion string, cni string) (*v1.SteveAPIObject, string, *apisV1.Cluster) {
```

Comment:
Instead of redoing all of this provisioning steps why don't you use this func? https://github.com/rancher/rancher/blob/4dc0be2fa4cfd1ee034346b2ca2a64eb4b99a81d/tests/v2/validation/provisioning/k3s/provisioning_node_driver.go#L27

Reply:
Right, I've made the required change to use the pre-written function now.